dynamiqs/solver.py

```diff
@@ -1,27 +1,42 @@
-from typing import ClassVar, Literal, Optional
+from __future__ import annotations
+
+from typing import ClassVar, Literal
 
-import diffrax as dx
 import equinox as eqx
 from jaxtyping import Scalar
 
 from .gradient import Adjoint, Autograd, Gradient
 
+NoneType = type(None)
+
 
 class Solver(eqx.Module):
-    SUPPORTED_GRADIENT: ClassVar[tuple[Gradient]] = ()
+    SUPPORTED_GRADIENT: ClassVar[tuple[Gradient | NoneType]] = (NoneType,)
```

Comment:
Is is true to say that all solvers can accept `None` as gradient? If so we could probably strip this `NoneType` type away

Reply:
It's true for now so let's remove it yes 👍🏼